libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -828,6 +828,25 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         )
     }
 
+    fun previewNavigationRoutes(
+        routes: List<NavigationRoute>,
+        initialLegIndex: Int = 0,
+    ) {
+        //TODO: parse alternatives metadata using NN and set it route alternatives controller
+        directionsSession.setRoutes(routes, initialLegIndex, RoutesExtra.ROUTES_UPDATE_REASON_NEW)
```

Comment:
`directionsSession` notifies `RouteObserver`s. From the docs:
```
     * The route at index 0, if exist, will be treated as the primary route for 'Active Guidance'.
     *
     * A list of routes can be modified internally and externally at any time with
     * [MapboxNavigation.setRoutes], or during automatic reroutes, faster route and route refresh operations.
```
Wouldn't it be too big a change in the behaviour? Because now it's not only about ActiveGuidance anymore. I'm not sure it would, just wondering.

Reply:
new state won't break existing behaviour because only new methods can trigger new state